lib/spack/spack/fetch_strategy.py

```diff
@@ -421,6 +430,79 @@ def fetch(self):
         tty.msg('Using cached archive: %s' % path)
 
 
+class PyPiFetchStrategy(URLFetchStrategy):
+    """Fetches packages directly from PyPi using XML-RPC to query the API."""
+
+    enabled = True
+    required_attributes = ['pypi']
+
+    def __init__(self, pypi=None, digest=None, **kwargs):
+        # If pypi or digest are provided in the kwargs, then prefer
+        # those values.
+        self.pypi = kwargs.get('pypi', None)
+        if not self.pypi:
+            self.pypi = pypi
+
+        self.version = kwargs.get('version', None)
+
+        self.digest = kwargs.get('md5', None)
+        if not self.digest:
+            self.digest = digest
+
+        self.expand_archive = kwargs.get('expand', True)
+        self.extra_curl_options = kwargs.get('curl_options', [])
```

Comment:
Same here (sort of).  Just be aware of this Python trap:
http://stackoverflow.com/questions/366422/what-is-the-pythonic-way-to-avoid-default-parameters-that-are-empty-lists
So I would do:
```
def __init__(..., curl_options=None)
    ...
    self.curl_options = (list() if curl_options is None else curl_options)
```
Using `list()` here is interchangeable with `[]`, but it does make it explicit that you are instantiating a new list at runtime.